### DISCUSSION TITLE
Is it possible to disable the SentryMessageHandler/SentryHttpMessageHandler?

### DISCUSSION THREAD

**Discussion by evolvedlight** [getsentry/sentry-dotnet#3671] [Q&A] [answered]
Hi,

When I add the Sentry client, it's automatically adding this handler to every HTTP Client. However, there's an access token endpoint I use that doesn't like the extra headers added and the request is dropped (a firewall like behaviour). Is it possible to disable this behavior in any way? Looking through the code I don't currently see any possibility of this?

Thanks,
Steve

**Comment by jamescrosswell** [comment]
Hey @evolvedlight,

Which headers are causing you problems? I'm thinking that they may be headers that Sentry requires...

**Comment by evolvedlight** [comment]
Hi @jamescrosswell 

Apologies for the slow response - I was maybe only 50% correct in what I said above: the issue is that my service also has to make a request to a oauth token endpoint that is unrelated to anything sentry. The issue is that this oauth provider has "hardened" their endpoint and it rejects any requests with headers over a certain size.
It's actually the opentelemetry headers that are being added that breaks this - the stuff in here:
https://github.com/getsentry/sentry-dotnet/blob/a18bef32a7c55b8331574e9bfdd25a976d8b4fef/src/Sentry/BaggageHeader.cs#L11
This baggage header gets quite large, the endpoint hardening/firewall solution rejects this large header, and thus my app can't get access tokens when this tracing is applied. 
I'd done this: 
c.TracesSampler = (_) => null;
But that didn't seem to work, oddly.

**Reply by jamescrosswell** [reply, marked as answer]
Ah I see... thanks @evolvedlight. Yes the baggage headers would be passed on regardless as we don't want to break downstream tracing in a scenario where you have Service A (tracing enabled - creates baggage headers) -> Service B (using .NET SDK) -> Service C (tracing enabled - wants baggage headers).

I assume you're using `Sentry.Extensions.Logging` since that appears the be the only integration that automatically adds the SentryMessageHandler (via an implementation of `IHttpMessageHandlerBuilderFilter`):
https://github.com/getsentry/sentry-dotnet/blob/dcc32b22b29fe77f1129f9cb848388af6093ece5/src/Sentry.Extensions.Logging/SentryHttpMessageHandlerBuilderFilter.cs#L17-L22

That gets added here:
https://github.com/getsentry/sentry-dotnet/blob/dcc32b22b29fe77f1129f9cb848388af6093ece5/src/Sentry.Extensions.Logging/Extensions/DependencyInjection/ServiceCollectionExtensions.cs#L44

So if you request an HttpClient from the Services collection, it will presumably build one that has a SentryMessageHandler.

Two options I could think of:
1. You could add your own `IHttpMessageHandlerBuilderFilter` that looks for the `SentryMessageHandler` and removes it... this would remove it for all requests though (maybe not what you want - that would break tracing everywhere)
2. Rather than asking for an instance of an HttpClient from the Services collection, you could just create one: `var httpClient = new HttpClient()`... that way you know it doesn't use a SentryMessageHandler and you could do that **_just_** for the OAuth requests that are causing you trouble.

**Reply by evolvedlight** [reply]
Thanks - makes sense.

**Comment by evolvedlight** [comment]
In the end, the actual problem was this: I wanted to disable the traces, but had this line:
`c.TracesSampler = (_) => null;`
It needed to be:
`c.TracesSampler = (_) => 0;`
